packages/pigment-css-react/README.md

```diff
@@ -831,25 +752,31 @@ function App() {
   return (
     <div>
       {[...Array(10)].map((_, index) => (
-        <div key={index} className={css`${generateBubbleVars()}`} />
+        <div
+          key={index}
+          className={css`
+            ${generateBubbleVars()}
+          `}
+        />
       ))}
     </div>
-  )
+  );
 }
 ```
 
-However, in Pigment CSS with the same code as above, all instances have the same styles and won't change between renders because the styles are extracted at build time.
+But if you wrote the code above with Pigment CSS, all instances would have the same styles that would never change between renders because they're extracted at build time.
 
-To achieve the same result, you need to move the dynamic logic to props and pass the value to CSS variables instead:
+To implement programmatically generated styles with Pigment CSS, you must move the dynamic logic to props and pass the value to CSS variables instead.
+The snippet below leads to values that change between renders and instances as intended in the prior example:
 
 ```js
 function randomBetween(min: number, max: number) {
   return Math.floor(Math.random() * (max - min + 1) + min);
 }
 
 const Bubble = styled('div')({
-  '--x': props => props.x,
-  '--y': props => props.y,
+  '--x': (props) => props.x,
```

Comment:
I think this is the first reference to `props => x` in the README. I'm a bit confused about this, let me try to clarify why:

Above, we have two sections:
- "Styling based on props" -> we explain this approach works great when values are known at build time. Gives the impression that accessing the `props` object is not possible in `styled`.
- "Styling based on runtime values" -> we explain two paths to handle runtime values:
  - Using hand-written inline CSS vars.
  - Using a callback: Pigment will transform it to an inline CSS var and inline the return value of the callback so it runs at runtime.

This section, "Programmatically generated styles", seems like another use case of "Styling based on runtime values". I expected to see one of the two solutions named above (inline CSS vars or callbacks). Instead, we say accessing `props` is possible, but the rest of the README gives the impression that it's not possible.

If using `props => value` callbacks is possible, should we document it in the "Styling based on runtime values" section above, too?

Reply:
We've already mentioned accessing props in the section `Using a callback` that you referenced. It's the same thing except there, we are destructuring `isError`, so there is no direct mention of props. But it's the same thing. See line 333.

Reply:
Oh, I see! I think more people will miss this. Maybe we can improve the "Using a callback" part to be more explicit about being able to access `props`:

- Reword the explanation under "Use a callback function as a value ..." so we explicitly mention that callbacks have access to `props`.
- Do not destructure `props` in the example so it's clear what's going on.
  ```diff
  const Heading = styled('h1')({
  -  color: ({ isError }) => (isError ? 'red' : 'black'),,
  +  color: (props) => (props.isError ? 'red' : 'black'),
  });
  ```
- Maybe put "**Option 2** – Use a callback ..." before "**Option 1** – Declare a CSS variable...". I think styling based on props is probably a more widespread use case than creating custom CSS vars.

---

Food for thought: right now, we make the distinction between "props" (in the "Styling based on props" section) and "runtime values" (in the "Styling based on runtime values"), but maybe the distinction could be better conveyed with "static" (build time) vs "dynamic" values (runtime), because runtime styling can also be based on `props` via callbacks.